timesketch/api/v1/resources/sigma.py

```diff
@@ -153,11 +217,371 @@ def post(self):
+            logger.error(error_msg)
+            abort(
+                HTTP_STATUS_CODE_CONFLICT,
+                error_msg,
+            )
+
+        return self.to_json(
+            sigma_rule_from_db, status_code=HTTP_STATUS_CODE_OK
+        )
+
+
+class SigmaRuleByTextResource(resources.ResourceMixin, Resource):
+    """Get a Sigma rule by providing a text."""
+
+    @login_required
+    def post(self):
+        """Get a parsed Sigma rule by providing a text
+
+        Will parse a provided text `rule_yaml`, parse it and return as JSON
+
+
+        If no form content is given, the method will abort with HTTP Error 400.
+
+        Returns:
+            JSON sigma rule e.g.
+                {"objects": [sigma_rule], "meta": {"parsed": True}}
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        content = form.get("content")
+        if not content:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing value in the request. Nothing to parse",
+            )
+
+        try:
+            sigma_rule = ts_sigma_lib.parse_sigma_rule_by_text(content)
+
+        except ValueError as e:
+            error_msg = (
+                "Sigma rule Parsing error with provided rule {0!s}".format(e)
+            )
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+
+        except NotImplementedError as e:
+            error_msg = (
+                "Sigma Parsing error: Feature in the rule provided"
+                " is not implemented in this backend {0!s}".format(e)
+            )
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+
+        except sigma_exceptions.SigmaParseError as e:
+            error_msg = "Sigma parsing error generating rule"
+            " with error: {0!s}".format(e)
+            logger.error(error_msg, exc_info=True)
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+
+        except yaml.parser.ParserError as e:
+            error_msg = (
+                "Sigma parsing error: invalid yaml provided {0!s}".format(e)
+            )
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+
+        if not sigma_rule:
+            error_msg = "Sigma parsing error: No sigma rule was parsed"
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(HTTP_STATUS_CODE_NOT_FOUND, error_msg)
```

Comment:
In which scenarios can this ever happen? It feels like with all the checks you're doing before, there should be a rule here. 

In any case, I wouldn't raise a 404 because it's not about a resource not being found, but about a request not being properly formatted. So something like 400 would be a bit better.